### ISSUE TITLE
solc gas output different from remix

### ISSUE THREAD

**Issue by PatrickAlphaC** [ethereum/remix-project#2539]
I have a file that looks as such:

```javascript
// SPDX-License-Identifier: MIT
pragma solidity 0.8.7;

contract BasicSolContract {
    uint256 public myVar;

    function updateVar(uint256 myNewVar) external {
        myVar = myNewVar;
    }
}
```

When I compile it with solc and look for a gas estimate, I get the following:

```
solc --gas --base-path . -o ./build ./contracts/solidity/BasicSolContract.sol

Gas estimation:
construction:
   111 + 61400 = 61511
external:
   myVar():     2429
   updateVar(uint256):  22498
```

However in remix, when I deploy it, I can see the gas is different.

<img width="743" alt="Screen Shot 2022-06-15 at 11 07 24 AM" src="https://user-images.githubusercontent.com/54278053/173861671-5578f2b4-69f1-49a7-9abe-53c94690ded5.png">

The optimizer is enabled with 200 runs in remix.

**Comment by Aniket-Engg** [comment]
@PatrickAlphaC Have you compared with other ways of sending transactions?

**Comment by Aniket-Engg** [comment]
I think this is not specific to Remix or VM. Without optimization, Remix shows gas cost as 119683 which is similar If contract is deployed on Goerli
![Screenshot 2022-06-28 at 12 42 25 PM](https://user-images.githubusercontent.com/30843294/176117659-e124fec5-2913-46b4-add8-df172359978b.png)
![Screenshot 2022-06-28 at 12 49 09 PM](https://user-images.githubusercontent.com/30843294/176117894-7e23ea70-58c5-4ce8-b9a7-adcabd984a9f.png)

**Comment by PatrickAlphaC** [comment]
Yeah... I'm not sure who is right. I'm thinking someone's compiler is doing something funky.

**Comment by Aniket-Engg** [comment]
@PatrickAlphaC I have asked Solidity team to have a look. Let's see.

**Comment by Aniket-Engg** [comment]
This is known issue: https://github.com/ethereum/solidity/issues/8920

**Comment by PatrickAlphaC** [comment]
Wow! Thanks for this!

**Comment by cameel** [comment]
Are there any alternative libraries that Remix could use for gas estimation? Honestly, there's so much that the estimator in the compiler does not cover that we're debating if it's even useful in practice. And even if we fix the constructor quirk, it's unlikely we'll be putting much more work into it in the near future...

**Comment by PatrickAlphaC** [comment]
It looks like the remix gas estimator is actually "good" right?

**Comment by cameel** [comment]
Ah, right. I mean, it's not estimation, it's the cost of actual execution, but yeah, the numbers from screenshots do not need fixing.

I just thought that Remix uses compiler's estimator somewhere but maybe I was wrong about that. I vaguely remember someone complaining about wrong estimations from Remix that in the end turned out to be compiler's estimations.
